src/server/game/Groups/Group.cpp

```diff
@@ -1494,6 +1494,19 @@ void Group::CountTheRoll(Rolls::iterator rollI, Map* allowedMap)
             item->is_blocked = false;
     }
 
+    {
+        Loot* loot = roll->getLoot();
+        if (loot->isLooted() && loot->sourceGameObject != nullptr)
+        {
+            const GameObjectTemplate* goInfo = loot->sourceGameObject->GetGOInfo();
+            if (goInfo != nullptr && goInfo->type == GAMEOBJECT_TYPE_CHEST)
```

Comment:
I believe you can do this as (goInfo && goInfo->....)

That should check for null without you needing to deny null

Reply:
Yes they're the same thing, I just like being explicit, I believe it makes code more readable

Reply:
I would prefer the other approach as it is something we already use a lot, to keep the same code standard